src/engine/Engine.hpp

```diff
@@ -140,10 +126,17 @@ class Engine {
     inline bool hasLayers() { return !m_layers.empty(); }
     inline bool hasActiveLayers() { return !m_activeLayers.empty(); }
 
-    // Drawables
     std::shared_ptr<Scene> m_scene;
+
+    enum DeferType {
+        Add = 1,
+        Delete,
+        Activate,
+        Deactivate,
+    };
```

Comment:
I know this will be useful in the event handler aswell, we should add this to a separate header file.

Reply:
Sure thing!